packages/components/src/templates/next/components/complex/Accordion/Accordion.tsx

```diff
@@ -1,17 +1,49 @@
-import { BiChevronDown } from "react-icons/bi"
+"use client"
 
-import type { AccordionProps } from "~/interfaces"
+import type { VariantProps } from "tailwind-variants"
+import { useState } from "react"
+import { BiMinus, BiPlus } from "react-icons/bi"
+
+import type { AccordionProps as BaseAccordionProps } from "~/interfaces"
+import { tv } from "~/lib/tv"
 import { Prose } from "../../native"
 
-const Accordion = ({ summary, details }: AccordionProps) => {
+const createAccordionStyles = tv({
+  slots: {
+    details:
+      "group border-y border-divider-medium px-4 py-5 has-[+_details]:border-b-0",
+    summary:
+      "prose-headline-lg-medium flex list-none flex-row items-center justify-between gap-3 text-base-content-strong outline-none outline outline-0 outline-offset-2 outline-brand-interaction hover:cursor-pointer focus-visible:outline-2",
+    icon: "h-6 w-6 flex-shrink-0",
+    content: "pt-5 text-base-content-strong",
+  },
+})
+
+const accordionStyles = createAccordionStyles()
+
+interface AccordionProps
+  extends BaseAccordionProps,
+    VariantProps<typeof createAccordionStyles> {
+  name?: string
+}
```

Comment:
question(blocking): this `name` prop cannot be modified directly by the user as it is not part of the schema. Is this setup in preparation for the accordion group?

Also, these props should be part of the interfaces folder so it is easier to abstract out when we split up the package in the future.

Reply:
yea, if we want to actually only allow one accordion to open at a time. But actually I'll probably remove this first since that is not an expected feature yet.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details#name